### ISSUE TITLE
Validation fails when an empty string is provided explicitly for a required string property

### ISSUE THREAD

**Issue by joshua-berry-ntnx** [swagger-model-validator/swagger-model-validator#137]
The Swagger data model documentation for a string says:

> Note that an empty string "" is a valid string unless minLength or pattern is specified.
> — https://swagger.io/docs/specification/data-models/data-types/#string

If I define a schema like so:

```
{
    type: 'object',
    required: ['foo'],
    properties: {
        foo: {type: 'string'},
    },
}
```

Then per the Swagger documentation, I would expect `{foo: ""}` to validate. However, it fails with "foo is a required field".

I took a quick look and you actually have a test case covering this, but based on the documentation above, I believe the test case (and thus the validator behavior) is wrong:

https://github.com/swagger-model-validator/swagger-model-validator/blob/master/tests/testStringValidation.js#L118

`errors.valid` should be true with no error messages.

**Comment by brodyd** [comment]
I think we can manage that.  I'll see what its doing soon.

**Comment by joshua-berry-ntnx** [comment]
Thanks, much appreciated! :)

**Comment by brodyd** [comment]
This is fixed in 3.0.19.  Thanks for letting me know.
